agent-operator/src/main/java/com/walmartlabs/concord/agentoperator/agent/AgentClientFactory.java

```diff
@@ -0,0 +1,55 @@
+package com.walmartlabs.concord.agentoperator.agent;
+
+/*-
+ * *****
+ * Concord
+ * -----
+ * Copyright (C) 2017 - 2024 Walmart Inc.
+ * -----
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =====
+ */
+
+import java.net.http.HttpClient;
+import java.time.Duration;
+
+import static java.net.http.HttpClient.Redirect.NEVER;
+
+public class AgentClientFactory {
+
+    private static final Duration CONNECT_TIMEOUT = Duration.ofSeconds(10);
+
+    private final boolean useMaintenanceMode;
+    private final HttpClient httpClient;
+
+    public AgentClientFactory(boolean useMaintenanceMode) {
+        this.useMaintenanceMode = useMaintenanceMode;
+        if (useMaintenanceMode) {
+            this.httpClient = HttpClient.newBuilder()
+                    .version(HttpClient.Version.HTTP_1_1)
+                    .followRedirects(NEVER)
+                    .connectTimeout(CONNECT_TIMEOUT)
+                    .build();
+        } else {
+            httpClient = null;
+        }
+    }
+
+    public AgentClient create(String podIp) {
```

Comment:
Maybe pass the entire Pod object here? :thinking: 
It's not like we need anything but the IP here (for now) but it feels safer to avoid naked Strings.